tests_e2e/orchestrator/lib/agent_test_suite.py

```diff
@@ -587,6 +599,8 @@ def _execute(self) -> None:
                     self._clean_up(test_suite_success and not unexpected_error)
                     if unexpected_error:
                         self._mark_log_as_failed()
+                    if not test_suite_success or unexpected_error:
```

Comment:
I'd just add a comment explaining the reason of this exception

Reply:
Updated